cmd/juju/application/resolved.go

```diff
@@ -0,0 +1,116 @@
+// Copyright 2012, 2013 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package application
+
+import (
+	"github.com/juju/cmd"
+	"github.com/juju/errors"
+	"github.com/juju/gnuflag"
+	"gopkg.in/juju/names.v2"
+
+	"github.com/juju/juju/api/application"
+	"github.com/juju/juju/cmd/juju/block"
+	"github.com/juju/juju/cmd/modelcmd"
+)
+
+func NewResolvedCommand() cmd.Command {
+	return modelcmd.Wrap(&resolvedCommand{})
+}
+
+// resolvedCommand marks a unit in an error state as ready to continue.
+type resolvedCommand struct {
+	modelcmd.ModelCommandBase
+	applicationResolveAPI applicationResolveAPI
+	clientAPI             clientAPI
+
+	UnitNames []string
+	NoRetry   bool
+	All       bool
+}
+
+func (c *resolvedCommand) Info() *cmd.Info {
+	return &cmd.Info{
+		Name:    "resolved",
+		Args:    "[<unit> ...]",
+		Purpose: "Marks unit errors resolved and re-executes failed hooks.",
+		Aliases: []string{"resolve"},
+	}
+}
+
+func (c *resolvedCommand) SetFlags(f *gnuflag.FlagSet) {
+	c.ModelCommandBase.SetFlags(f)
+	f.BoolVar(&c.NoRetry, "no-retry", false, "Do not re-execute failed hooks on the unit")
+	f.BoolVar(&c.All, "all", false, "Marks all units in error as resolved")
```

Comment:
I don't think we should land this branch as it is. We either support --all in the apiserver first, or we don't expose the --all flag on this command until server side is done.

Reply:
I'll hide the --all on the CLI